### PR TITLE
[SYCL] Rewrite image_api LIT as Google Unit test

### PR DESCRIPTION
**Pull request by alexbatashev**
Avoids usage of private APIs in end-to-end tests, places a scheduler
test alongside its siblings. Also fixes a few warnings revealed by
this refactoring.

Signed-off-by: Alexander Batashev alexander.batashev@intel.com

### CODE REVIEW

File: sycl/unittests/scheduler/ImageApi.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  constexpr sycl::image_channel_type ChannelType =
+      sycl::image_channel_type::fp32;
+  const sycl::range<Dimensions> Range{16, 16};
+
+  sycl::image<Dimensions> SrcImg(Src.data(), ChannelOrder, ChannelType, Range);
+  sycl::image<Dimensions> DstImg(Dest.data(), ChannelOrder, ChannelType, Range);
+
+  auto CGHLambda = [&](MockHandler &CGH) {
+    auto SrcAcc =
+        SrcImg.template get_access<sycl::float4, sycl::access::mode::read>(CGH);
```
Comment (romanovvlad) [inline sycl/unittests/scheduler/ImageApi.cpp]: I'm not sure this test makes sense when it doesn't use internal API.
Why can't we continue using internal API in google tests?
And if we cannot, I suggest removing the test completely.

Comment (alexbatashev) [reply sycl/unittests/scheduler/ImageApi.cpp]: By "use internal API" do you mean creating accessor directly inside the handler?
I think it is a duplication of code. And if the logic inside the handler changes a bit, this test will become invalid (yet it still may pass). Do we need such a test?

Comment (romanovvlad) [reply sycl/unittests/scheduler/ImageApi.cpp]: Let's remove it.

**Comment by alexbatashev** [comment]
Replaced with #1349

**Pull request closed without merging**